src/Generator/Generators/CSharp/CSharpMarshal.cs

```diff
@@ -501,8 +501,10 @@ public override bool VisitPointerType(PointerType pointer, TypeQualifiers quals)
             {
                 if (Context.Parameter.Usage == ParameterUsage.Out)
                 {
+                    var TypePrinter = new CSharpTypePrinter(Context.Driver);
```

Comment:
The marshaller already has a type printer contained in a field so please use that instead of creating a new one.